loadable_apps/loadable_sample/wifiapp/binary_update.c

```diff
@@ -148,16 +155,20 @@ static int binary_update_download_binary(binary_update_info_t *binary_info, int
 	ret = lseek(write_fd, 0, SEEK_SET);
 	if (ret != 0) {
 		printf("Failed to lseek %d, errno %d\n", ret, errno);
+		unlink(new_filepath);
+		ret = ERROR;
 		goto errout_with_close_fd2;
 	}
 	ret = write(write_fd, (FAR uint8_t *)&crc_hash, CHECKSUM_SIZE);
 	if (ret != CHECKSUM_SIZE) {
 		printf("Failed to write %d\n", ret);
+		unlink(new_filepath);
```

Comment:
1. Don't you need this for ok case?
2. Is it ok to call this before closing the fd?